gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/converter/HiveAvroToOrcConverter.java

```diff
@@ -122,27 +136,37 @@ public Schema convertSchema(Schema inputSchema, WorkUnitState workUnit) throws S
     return new SingleRecordIterable<>(conversionEntity);
   }
 
+  private boolean shouldFlattenSchema(WorkUnitState workUnit) {
```

Comment:
As discussed, its a bit of a necessity due to Converter interface. But a constant time operation, so livable.